### PR TITLE
AdvLoggerPkg: Add AdvancedLoggerMmCoreGoogleTest

### PR DESCRIPTION
**Pull request by VivianNK**
## Description
Add unit test for AdvancedLoggerLib for Mm Core.

Moved changes to the library to this PR: https://github.com/microsoft/mu_plus/pull/582
This PR is dependent on those changes and will be rebased after 582 is merged. 

---

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [x] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

CI

## Integration Instructions

N/A

### CODE REVIEW

File: AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  BOOLEAN status;
+  ADVANCED_LOGGER_INFO testLoggerInfo;
+  // StrictMock<MockHobLib> gHobLib;
+  // StrictMock<MockAdvancedLoggerHdwPortLib> gALHdwPortLib;
+
+  void
+  SetUp (
+    ) override
+  {
+    CHAR8  OutputBuf[] = "MyUnitTestLog";
```
Comment (apop5) [inline AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp]: same problem here with local variable as in other test.

File: AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/AdvancedLoggerLib.c [hunk 2]
```diff
@@ -7,7 +7,7 @@
 **/
 
 #include <PiDxe.h>
-
+#include <PiMm.h>
```
Comment (os-d) [inline AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/AdvancedLoggerLib.c]: Can we split this up into a REBASE & FF PR with the actual lib changes and the unit test separated?

File: AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/AdvancedLoggerLib.c [hunk 3]
```diff
@@ -88,29 +89,28 @@ AdvancedLoggerGetLoggerInfo (
...
-    GuidHob = GetFirstGuidHob (&gAdvancedLoggerHobGuid);
+    GuidHob = (EFI_HOB_GUID_TYPE *)GetFirstGuidHob (&gAdvancedLoggerHobGuid);
     if (GuidHob == NULL) {
       DEBUG ((DEBUG_ERROR, "%a: Advanced Logger Hob not found\n", __FUNCTION__));
     } else {
       LogPtr      = (ADVANCED_LOGGER_PTR *)GET_GUID_HOB_DATA (GuidHob);
       mLoggerInfo = ALI_FROM_PA (LogPtr->LogBuffer);
       if (mLoggerInfo != NULL) {
         mMaxAddress = LOG_MAX_ADDRESS (mLoggerInfo);
+        DEBUG ((DEBUG_INFO, "%a: LoggerInfo=%p\n", __FUNCTION__, mLoggerInfo));
```
Comment (os-d) [inline AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/AdvancedLoggerLib.c]: while you are changing this, may as well go to `__func__`. Also, this is the one instance where `%a` is not that helpful on its own because we have so many of these libs that print where logger info is. We should probable add which module it is too (that could be a separate PR adding to each lib)

File: AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  EXPECT_EQ (Status, FALSE);
+
+  // Invalid Signature
+  mLoggerInfo            = &testLoggerInfo;
+  mLoggerInfo->Signature = SIGNATURE_32 ('T', 'E', 'S', 'T');
+  Status                 = ValidateInfoBlock ();
+  EXPECT_EQ (Status, FALSE);
+  mLoggerInfo->Signature = ADVANCED_LOGGER_SIGNATURE;
+
+  // Mismatched Version is okay? Wouldn't expect mismatched version with valid signature?
```
Comment (os-d) [inline AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp]: We should check Version, too. The signature tells us we have the right structure (well, hopefully), the version may be diferent if this MMCore instance is sharing a log buffer with DXE that thinks it is different, etc.

File: AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+  // Invalid Buffer Size
+  mLoggerInfo->LogBufferSize = (UINT32)0;
+  mBufferSize                = (UINT32)0x1000;
+  Status                     = ValidateInfoBlock ();
+  EXPECT_EQ (Status, FALSE);
+}
+
+/*/* Commented out, need mock libraries to be implemented.
+// Test AdvancedLoggerGetLoggerInfo NULL HOB
+TEST_F (AdvancedLoggerMmCoreTest, AdvLoggerGetInfoNullHob) {
```
Comment (os-d) [inline AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp:107]: Are these mocks ready to go?

File: AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/AdvancedLoggerLib.c (hunk 2 above)
Comment (VivianNK) [reply AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/AdvancedLoggerLib.c]: https://github.com/microsoft/mu_plus/pull/582

File: AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp (hunk 4 above)
Comment (VivianNK) [reply AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp]: Made change in https://github.com/microsoft/mu_plus/pull/582 to add a version check, will update the test.

File: AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp (hunk 5 above)
Comment (VivianNK) [reply AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/GoogleTest/AdvancedLoggerMmCoreGoogleTest.cpp:107]: Talked offline. They are not ready.

File: AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/AdvancedLoggerLib.c (hunk 2 above)
Comment (VivianNK) [reply AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/AdvancedLoggerLib.c]: Merged.

**Review by os-d — APPROVED** [review]

**Review by apop5 — APPROVED** [review]
